### PR TITLE
Added targets enum table, restructured config

### PR DESCRIPTION
**Pull request by guderkar**

### CODE REVIEW

File: src/featurestorebundle/widgets/WidgetsFactory.py [hunk 1]
```diff
@@ -37,19 +40,13 @@ def create_for_target(self):
         self.__widgets.add_text("number_of_time_units", self.__defaults.number_of_time_units)
 
     def create_target_name(self):
+        targets = [
+            row.__getattr__(get_target_id_column_name())
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/widgets/WidgetsFactory.py:44]: why not `getattr(row, get_target_id_column_name())` ?

File: src/featurestorebundle/widgets/WidgetsFactory.py [hunk 2]
```diff
@@ -37,19 +40,13 @@ def create_for_target(self):
...
+            for row in self.__targets_reader.read_enum().select(get_target_id_column_name()).collect()
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/widgets/WidgetsFactory.py:45]: does this run every time a notebook is run?

File: src/featurestorebundle/delta/target/reader/DeltaTableTargetsReader.py [hunk 3]
```diff
@@ -16,9 +16,14 @@ def __init__(
...
 
-    def read(self, entity: str) -> DataFrame:
-        full_table_name = self.__table_names.get_entity_targets_full_table_name(entity)
+    def read(self, entity_name: str) -> DataFrame:
+        full_table_name = self.__table_names.get_targets_full_table_name(entity_name)
 
-        self.__logger.info(f"Reading targets for entity {entity} from table {full_table_name}")
+        self.__logger.info(f"Reading targets for entity {entity_name} from table {full_table_name}")
+
+        return self.__spark.read.table(full_table_name)
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/delta/target/reader/DeltaTableTargetsReader.py:24]: so target_store has to be a table? not a delta file?

Comment (guderkar) [reply src/featurestorebundle/delta/target/reader/DeltaTableTargetsReader.py:24]: Can be delta file, it has configurable backend

File: src/featurestorebundle/widgets/WidgetsFactory.py (hunk 2 above)
Comment (guderkar) [reply src/featurestorebundle/widgets/WidgetsFactory.py:45]: even every time you you changes widgets but its pretty fast as the table is small

File: src/featurestorebundle/widgets/WidgetsFactory.py (hunk 1 above)
Comment (guderkar) [reply src/featurestorebundle/widgets/WidgetsFactory.py:44]: fair point

**Pull request merged**